# PointNet++ Training experimental

This notebook serves the purpose of creating the training routine for the PointNet++ to encode point clouds into the same latent space as the CAD sequence encoding done by the DeepCAD model.

The pointnet venv is located at "Point-Cloud-Reconstruction/pointnet.pytorch", but the model used is in "Point-Cloud-Reconstruction/Pointnet_Pointnet2_pytorch"

## 1. Check point cloud data
We will use open3d to inspect the point cloud files. During the conversion of the CAD sequences to point clouds errors appeared, which is why I assume there must be some corrupt point cloud files, which need to be excluded.

- Load train/val/test split
- create train/val/test pc path lists

In [130]:
import open3d as o3d
import os
from tqdm import tqdm

In [131]:
DATA_ROOT = "../data/pc_cad"
SPLIT = "../data/train_val_test_split.json"

In [132]:
with open(SPLIT, "r") as fp:
    all_data = json.load(fp)
print(f"Number of samples that should be in the split: {len(train)+len(val)+len(test)}")
for phase in all_data.keys():
    print(phase, len(all_data[phase]))

Number of samples that should be in the split: 178238
train 161240
validation 8946
test 8052


In [133]:
train = [os.path.join(DATA_ROOT, f"{idx}.ply") for idx in all_data['train']]
val = [os.path.join(DATA_ROOT, f"{idx}.ply") for idx in all_data['validation']]
test = [os.path.join(DATA_ROOT, f"{idx}.ply") for idx in all_data['test']]

Now we need to check if the point cloud files in the split are actually there and not corrupt.

In [134]:
from glob import glob
directory = DATA_ROOT
file_pattern = "**/*.ply"
all_files = glob(f"{directory}/{file_pattern}", recursive=True)
print(f"Total number of files: {len(all_files)}")
print(f"Missing/corrupt files: {len(train)+len(val)+len(test)-len(all_files)}")

Total number of files: 177948
Missing/corrupt files: 290


Apparently the conversion from json to point cloud did not work in 290 cases. These have to be removed from the split.

In [144]:
corrupt_files = []
correct_files = []
for pc_file in train:
    try:
        pc = o3d.io.read_point_cloud(pc_file)
        if not pc.has_points():
            corrupt_files.append(pc_file)
    except Exception as e:
        corrupt_files.append(pc_file)
    if not os.path.exists(pc_file):
        corrupt_files.append(pc_file)
    else:
        correct_files.append(pc_file)

RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0086/00866760.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0022/00222961.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0083/00837927.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0070/00704820.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0087/00870042.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0070/00703301.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0081/00817507.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0033/00333559.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0058/00581385.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0082/00827723.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0037/00377294.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0029/00299584.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00787215.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00780606.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0022/00229900.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0076/00763366.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0035/00352033.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0071/00710683.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0061/00619072.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0072/00726035.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0045/00453232.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0068/00682204.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0057/00576824.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00407728.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0056/00568170.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0019/00197175.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0095/00957222.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0013/00132466.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0094/00944573.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0071/00718882.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0057/00576294.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0069/00692559.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0041/00414871.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0056/00562023.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0090/00907050.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0095/00959367.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0035/00355362.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0030/00309152.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0008/00088295.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0004/00040361.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0086/00864537.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00780510.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00784839.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0025/00258885.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0034/00346987.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0049/00499619.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0070/00701017.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0076/00769895.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0075/00758550.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0044/00445404.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0080/00808088.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0037/00372967.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0064/00643667.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0067/00674424.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0098/00980295.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0066/00665039.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0044/00441717.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0070/00706827.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0093/00931985.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0066/00660519.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0010/00101487.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0035/00355496.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0008/00082886.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0008/00085817.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0097/00974972.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0024/00247640.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0007/00078377.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0075/00750356.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0066/00663833.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0089/00897011.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0009/00095310.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00783425.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0048/00484266.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0079/00790669.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0058/00585548.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0052/00523137.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0071/00711118.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0065/00657014.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0066/00666904.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0058/00588563.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0019/00191878.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0095/00952815.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0031/00312964.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0071/00715087.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0082/00825171.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0005/00052232.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00404561.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0081/00817492.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0027/00272044.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0066/00666947.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0055/00554853.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0066/00663822.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0060/00609074.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00781749.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0073/00730662.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0085/00853365.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0056/00565611.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0067/00673369.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0081/00811497.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00785524.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0099/00991995.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0021/00216313.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0050/00505737.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0081/00815219.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0075/00753984.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0056/00569390.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0034/00344562.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0074/00749974.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0068/00684122.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0041/00417782.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0071/00710645.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0020/00200578.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0060/00604318.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00783629.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0062/00629603.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0082/00827264.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0068/00683588.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0053/00530054.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0030/00300036.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0002/00024437.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0011/00116212.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0094/00945219.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0052/00528213.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00402915.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0037/00372974.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0047/00474797.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0035/00352166.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0097/00971985.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0045/00458677.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0012/00120848.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00408502.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0017/00174113.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0085/00852839.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0037/00372261.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0003/00039794.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0025/00258555.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0013/00132516.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0085/00855514.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00781137.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0004/00042529.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0006/00061450.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0002/00022522.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0070/00702943.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0048/00486544.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0034/00346497.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0068/00683717.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0087/00878518.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0095/00954461.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00404562.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0053/00533088.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0048/00488490.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0058/00580520.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0065/00656191.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0020/00203648.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0071/00711819.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0079/00790613.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0081/00812040.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0031/00318551.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0069/00695403.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0034/00340636.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0033/00338711.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0057/00573896.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0057/00573897.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0076/00766122.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0038/00389446.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0019/00198487.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0090/00900840.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0025/00256060.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0061/00610270.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0029/00299548.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0013/00130497.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0095/00951529.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0030/00309154.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00404563.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0080/00801588.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0059/00595308.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00404567.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0013/00132457.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0056/00569243.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0039/00395122.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00788448.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0096/00968673.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0027/00278062.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0020/00200893.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0057/00571083.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0009/00094794.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0094/00949646.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0013/00132468.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0026/00264314.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0034/00346714.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0085/00853155.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0045/00459362.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0092/00923521.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0066/00664130.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0041/00415895.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0038/00387617.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0027/00270361.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0041/00415456.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0056/00560203.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00787173.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0077/00777387.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0060/00604130.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0010/00104430.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0062/00620731.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0068/00681457.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0008/00081517.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0009/00093790.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0042/00423720.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0004/00041346.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0076/00765224.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0080/00809010.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0097/00977682.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0050/00509182.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0026/00265399.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00404570.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0011/00115554.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0062/00622338.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0027/00275536.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00404564.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0002/00022802.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0071/00710682.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0072/00722092.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0094/00947886.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0096/00965596.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0049/00497104.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0007/00077512.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0094/00943461.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0023/00235163.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0046/00464313.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0038/00387739.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0046/00463241.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0007/00076677.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0081/00811454.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0078/00783507.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0014/00146114.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0090/00906598.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0072/00724949.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0012/00124179.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0056/00560838.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0010/00107723.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0034/00342977.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0090/00900095.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0006/00064608.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0086/00868162.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0013/00139529.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0074/00743409.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0016/00160653.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0040/00405756.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0080/00808603.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0035/00359894.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0065/00653305.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0088/00880394.ply


RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0074/00743986.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0064/00649141.ply


RPly: Unable to open file
RPly: Unable to open file
RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0043/00438205.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0044/00445989.ply
[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0028/00284028.ply


RPly: Unable to open file


[Open3D WARNING] Read PLY failed: unable to open file: ../data/pc_cad/0046/00463296.ply


In [145]:
len(corrupt_files), len(correct_files)

(516, 160982)

Corrupt files in (test: 28, val: 36, train: 516) -> Total = 290 anscheinend mehr?? Ja anscheinend gab es 290 Fälle in denen nicht konvertiert werden konnte, aber noch ein paar mehr Fälle wo zwar konvertiert wurde, aber die point cloud an sich halt corrupt ist.

NEXT TIME: Herausfinden, welche PC corrupt ist und welche nicht konvertiert werden konnten (290), damit sollten (516-290) = 226 corrupt sein. Dann eine fertige liste mit funktionierenden point clouds erstellen. Am ende zum beispiel ein script "prepare pointnet++ training data.py" oder sowas erstellen.

In [107]:
print(no_points)

['../data/pc_cad/0056/00566337.ply']


Test: 14 pc's waren entweder point cloud conversion fail oder create cad fail, alle anderen korrekt
Val: 18 pc's waren entweder point cloud conversion fail oder create cad fail, alle anderen korrekt
Train: Irgendwo findet ein Parallels problem statt (0, 78950, 1, 82290)Es ist die file 0011/00116212

Error kommt von process_one -> create_CAD (in visualize.py)

Laut stackoverflow soll man mit ulimit -s stack anschauen und mit ulimit -s \<neuerWert> erhöhen

Chat GPT: "On macOS (and other Unix-like operating systems), the ulimit -s command is used to query or set the stack size limit for processes. The stack size determines the amount of memory allocated for a program's stack, which is used for function calls, local variables, and control flow."



In [113]:
problematic_file = "/Users/saidharb/Documents/LocalDocuments/Master-Thesis/Point-Cloud-Reconstruction/data/cad_json/0011/00116212.json"
try:
    with open(problematic_file, "r") as f:
        data = json.load(f)
    print("File loaded successfully.")
    print(json.dumps(data, indent=4))  # Pretty-print the JSON content.
except json.JSONDecodeError as e:
    print(f"JSON decoding error: {e}")
except Exception as e:
    print(f"Error reading file: {e}")

File loaded successfully.
{
    "entities": {
        "FRAcwEqNExbvyOz_2": {
            "name": "Extrude 3",
            "type": "ExtrudeFeature",
            "profiles": [
                {
                    "profile": "JNC",
                    "sketch": "FRSAyqbsz52iKa9_2"
                }
            ],
            "extent_two": {
                "distance": {
                    "type": "ModelParameter",
                    "role": "AgainstDistance",
                    "name": "none",
                    "value": 0.0
                },
                "type": "DistanceExtentDefinition",
                "taper_angle": {
                    "type": "ModelParameter",
                    "role": "Side2TaperAngle",
                    "name": "none",
                    "value": 0.0
                }
            },
            "extent_one": {
                "distance": {
                    "type": "ModelParameter",
                    "role": "AlongDistance",
                   

In [ ]:
from OCC.Core.BRepCheck import BRepCheck_Analyzer

def is_valid_shape(shape):
    analyzer = BRepCheck_Analyzer(shape)
    return analyzer.IsValid()
